swaps/swaps_service.go

```diff
@@ -0,0 +1,270 @@
+				return
+			}
+		}
+	}()
+
+	svc.cancelFn = cancelFn
+
+	return nil
+}
+
+func (svc *swapsService) StopAutoSwaps() {
+	if svc.cancelFn != nil {
+		logger.Logger.Info("Stopping swap service...")
+		svc.cancelFn()
+		logger.Logger.Info("swap service stopped")
+	}
+}
+
+func (svc *swapsService) ReverseSwap(ctx context.Context, amount uint64, destination string, lnClient lnclient.LNClient) error {
+	var network, err = boltz.ParseChain(svc.cfg.GetEnv().LDKNetwork)
+	if err != nil {
+		return err
+	}
+
+	ourKeys, err := btcec.NewPrivateKey()
+	if err != nil {
+		return err
+	}
+
+	preimage := make([]byte, 32)
+	_, err = rand.Read(preimage)
+	if err != nil {
+		return err
+	}
+	preimageHash := sha256.Sum256(preimage)
+
+	boltzApi := &boltz.Api{URL: svc.cfg.GetEnv().BoltzApi}
+
+	swap, err := boltzApi.CreateReverseSwap(boltz.CreateReverseSwapRequest{
+		From:           boltz.CurrencyBtc,
+		To:             boltz.CurrencyBtc,
+		ClaimPublicKey: ourKeys.PubKey().SerializeCompressed(),
+		PreimageHash:   preimageHash[:],
+		InvoiceAmount:  amount,
+	})
+	if err != nil {
+		return fmt.Errorf("could not create swap: %s", err)
+	}
+
+	boltzPubKey, err := btcec.ParsePubKey(swap.RefundPublicKey)
+	if err != nil {
+		return err
+	}
+
+	tree := swap.SwapTree.Deserialize()
+	if err := tree.Init(boltz.CurrencyBtc, true, ourKeys, boltzPubKey); err != nil {
+		return err
+	}
+
+	if err := tree.Check(boltz.ReverseSwap, swap.TimeoutBlockHeight, preimageHash[:]); err != nil {
+		return err
+	}
+
+	logger.Logger.WithField("swap", swap).Info("Swap created")
+
+	boltzWs := boltzApi.NewWebsocket()
+	if err := boltzWs.Connect(); err != nil {
+		return fmt.Errorf("could not connect to Boltz websocket: %w", err)
+	}
+
+	if err := boltzWs.Subscribe([]string{swap.Id}); err != nil {
+		return err
+	}
+
+	for update := range boltzWs.Updates {
+		parsedStatus := boltz.ParseEvent(update.Status)
+
+		switch parsedStatus {
+		case boltz.SwapCreated:
+			logger.Logger.WithFields(logrus.Fields{
+				"swap":   swap,
+				"update": update,
+			}).Info("Swap created, paying the invoice")
+			metadata := map[string]interface{}{
+				"swapId":        swap.Id,
+				"onchainAmount": swap.OnchainAmount,
+				"refundPubkey":  swap.RefundPublicKey,
+			}
+			go func() {
+				payCtx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+				defer cancel()
+				transaction, err := svc.transactionsService.SendPaymentSync(payCtx, swap.Invoice, nil, metadata, lnClient, nil, nil)
+				logger.Logger.WithFields(logrus.Fields{
+					"transaction": transaction,
+				}).Info("Swap created, paying the invoice")
+				if err != nil {
+					logger.Logger.WithError(err).WithFields(logrus.Fields{
+						"swap":   swap,
+						"update": update,
+					}).Error("Error paying the swap invoice")
+					return
+				}
+				logger.Logger.WithField("transaction", transaction).Info("Swap payment succeeded")
+			}()
+			break
+
+		case boltz.TransactionMempool:
```

Comment:
You are currently always accepting 0-conf from boltz, which leaves you to the potential risk of boltz replacing its lockup transaction after you revealed the preimage by broadcasting the claim transaction. Since the swaps arent exposed in any user interface you can always wait for `boltz.TransactionConfirmed` to make sure boltz could not replace the transaction. You can still leave a log statement for `boltz.TransactionMempool` but only claim once its confirmed.
```suggestion
		case boltz.TransactionMempool:
```
```suggestion
		case boltz.TransactionConfirmed:
```

Reply:
Nice, thanks!